File: topics/control-flow/exercises/ColorPick1.pas
procedure DrawColoredRectsV2();
begin
  repeat
    ProcessEvents();
    ClearScreen();
    
    if KeyDown(vk_R) then FillRectangle(ColorRed, 10, 10, 780, 580)
    else if KeyDown(vk_G) then FillRectangle(ColorGreen, 20, 20, 760, 560)
    else if KeyDown(vk_B) then FillRectangle(ColorBlue, 30, 30, 740, 540)
    else FillRectangle(ColorGrey, 40, 40, 720, 520);

    RefreshScreen();
  until KeyTyped(vk_ESCAPE) or WindowCloseRequested();
end;

File: topics/control-flow/exercises/color_pick1.c
void draw_colored_rects_v2()
{
    do
    {
        process_events();
        clear_screen();
        
        if (key_down(VK_R)) fill_rectangle(ColorRed, 10, 10, 780, 580);
        else if (key_down(VK_G)) fill_rectangle(ColorGreen, 20, 20, 760, 560);
        else if (key_down(VK_B)) fill_rectangle(ColorBlue, 30, 30, 740, 540);
        else fill_rectangle(ColorGrey, 40, 40, 720, 520);
        
        refresh_screen();
    } while ( ! ( key_typed(VK_ESCAPE) || window_close_requested() ) );
}